src/main/resources/hudson/plugins/promoted_builds/PromotedBuildAction/badge.jelly

```diff
@@ -1,11 +1,13 @@
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
   <j:if test="${it.hasPromotion()}">
     <j:forEach var="status" items="${it.getPromotions()}">
-      <a href="${link}promotion/">
-        <img width="16" height="16"
-          title="${status.name}"
-          src="${rootURL}${status.getIcon('16x16')}"/>
-      </a>
+      <j:if test="${status.getProcess().isVisible()}">
```

Comment:
<code>getProcess()</code> may return null if somebody deletes the process from the job config. https://github.com/jenkinsci/promoted-builds-plugin/blob/069d9fd258ecf89377738c571589bb8bee6ae68e/src/main/java/hudson/plugins/promoted_builds/Status.java#L96-L103